extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/authorization/db/cache/DefaultBasicAuthorizerCacheManager.java

```diff
@@ -0,0 +1,236 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.security.basic.authorization.db.cache;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Preconditions;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.metamx.emitter.EmittingLogger;
+import com.metamx.http.client.Request;
+import io.druid.client.coordinator.Coordinator;
+import io.druid.concurrent.LifecycleLock;
+import io.druid.discovery.DruidLeaderClient;
+import io.druid.guice.ManageLifecycle;
+import io.druid.guice.annotations.Smile;
+import io.druid.java.util.common.ISE;
+import io.druid.java.util.common.RetryUtils;
+import io.druid.java.util.common.StringUtils;
+import io.druid.java.util.common.concurrent.Execs;
+import io.druid.java.util.common.concurrent.ScheduledExecutors;
+import io.druid.java.util.common.lifecycle.LifecycleStart;
+import io.druid.security.basic.authentication.BytesFullResponseHandler;
+import io.druid.security.basic.authentication.BytesFullResponseHolder;
+import io.druid.security.basic.BasicAuthCommonCacheConfig;
+import io.druid.security.basic.authorization.BasicRoleBasedAuthorizer;
+import io.druid.security.basic.authorization.db.updater.CoordinatorBasicAuthorizerMetadataStorageUpdater;
+import io.druid.security.basic.authorization.entity.BasicAuthorizerRole;
+import io.druid.security.basic.authorization.entity.BasicAuthorizerUser;
+import io.druid.security.basic.authorization.entity.UserAndRoleMap;
+import io.druid.server.security.Authorizer;
+import io.druid.server.security.AuthorizerMapper;
+import org.jboss.netty.handler.codec.http.HttpMethod;
+import org.joda.time.Duration;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+@ManageLifecycle
+public class DefaultBasicAuthorizerCacheManager implements BasicAuthorizerCacheManager
+{
+  private static final EmittingLogger LOG = new EmittingLogger(DefaultBasicAuthorizerCacheManager.class);
+
+  private final ConcurrentHashMap<String, Map<String, BasicAuthorizerUser>> cachedUserMaps;
+  private final ConcurrentHashMap<String, Map<String, BasicAuthorizerRole>> cachedRoleMaps;
+
+  private final Set<String> authorizerPrefixes;
+  private final Injector injector;
+  private final ObjectMapper objectMapper;
+  private final LifecycleLock lifecycleLock = new LifecycleLock();
+  private final DruidLeaderClient druidLeaderClient;
+  private final BasicAuthCommonCacheConfig commonCacheConfig;
+
+  private volatile ScheduledExecutorService exec;
```

Comment:
Why is this volatile? And it should be shut down on stop().

Reply:
Fixed this, similar to previous comments